apps/src/studio/studio.js

```diff
@@ -4330,6 +4393,18 @@ var checkFinished = function () {
     return true;
   }
 
+  if (hasRequiredForSuccess) {
+    if (requiredForSuccess.achieved) {
+      Studio.message = null;
+      Studio.result = ResultType.SUCCESS;
+      return true;
+    } else {
+      // Not meeting these is not reason for immediate failure in itself, but they do
+      // establish a custom error message.
+      Studio.message = requiredForSuccess.message;
```

Comment:
This is a really good point; it'd be great to be uploading all of these results for future analysis.  I've opened pivotal https://www.pivotaltracker.com/story/show/105498648 to track this.  @cpirich we can chat more about this too, because I think you had similar concerns about integrating better with the existing system.